lib/iris/coords.py

```diff
@@ -1373,33 +1446,26 @@ def from_regular(cls, zeroth, step, count, standard_name=None,
             bounds values will be defined. Defaults to False.
 
         """
-        coord = DimCoord.__new__(cls)
-
-        coord.standard_name = standard_name
-        coord.long_name = long_name
-        coord.var_name = var_name
-        coord.units = units
-        coord.attributes = attributes
-        coord.coord_system = coord_system
-        coord.circular = circular
-
-        points = (zeroth+step) + step*np.arange(count, dtype=np.float32)
+        points = (zeroth + step) + step * np.arange(count, dtype=np.float32)
+        # Spoof `iris.util.is_regular`, which only works on coords (which we
+        # don't yet have).
+        diffs = np.diff(points)
+        if not np.allclose(diffs, np.mean(diffs), rtol=0.001):
+            points = (zeroth + step) + step * np.arange(count,
+                                                        dtype=np.float64)
```

Comment:
> how could it be?

Mmm, fair point, which completely misses the more fundamental point that the problem would have been lessened if this method had just called the coord constructor in the first place, which remains the most reasonable way for this method to work.

> factoring out a private method which does the duplicated functionality and which is called by both this method and the `__init__` method

But what's the point?! You're still duplicating the functionality of the `__init__`. And worse, you're proposing adding another method (that we don't need), and getting the constructor calling methods rather than just setting instance attributes. [Let the constructor be the constructor](https://goo.gl/images/8icfDc).